test/e2e/tests/petnames-signature-request.spec.js

```diff
@@ -0,0 +1,232 @@
+// TODO(dbrans): When petnames is fully enabled, make this file the new signature-request.spec.js.
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  regularDelayMs,
+  openDapp,
+  DAPP_URL,
+  defaultGanacheOptions,
+  unlockWallet,
+  WINDOW_TITLES,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+const signatureRequestType = {
+  signTypedData: 'Sign Typed Data',
+  signTypedDataV3: 'Sign Typed Data V3',
+  signTypedDataV4: 'Sign Typed Data V4',
+};
+
+const testData = [
+  {
+    type: signatureRequestType.signTypedData,
+    buttonId: '#signTypedData',
+    verifyId: '#signTypedDataVerify',
+    verifyResultId: '#signTypedDataVerifyResult',
+    expectedMessage: 'Hi, Alice!',
+    verifyAndAssertMessage: {
+      titleClass: '.request-signature__content__title',
+      originClass: '.request-signature__origin',
+      messageClass: '.request-signature__row-value',
+    },
+  },
+  {
+    type: signatureRequestType.signTypedDataV3,
+    buttonId: '#signTypedDataV3',
+    verifyId: '#signTypedDataV3Verify',
+    verifyResultId: '#signTypedDataV3VerifyResult',
+    expectedMessage: 'Hello, Bob!',
+    verifyAndAssertMessage: {
+      titleClass: '.signature-request__content__title',
+      originClass: '.signature-request__origin',
+      messageClass: '.signature-request-data__node__value',
+    },
+  },
+  {
+    type: signatureRequestType.signTypedDataV4,
+    buttonId: '#signTypedDataV4',
+    verifyId: '#signTypedDataV4Verify',
+    verifyResultId: '#signTypedDataV4VerifyResult',
+    expectedMessage: 'Hello, Bob!',
+    verifyAndAssertMessage: {
+      titleClass: '.signature-request__content__title',
+      originClass: '.signature-request__origin',
+      messageClass: '.signature-request-data__node__value',
+    },
+  },
+];
+
+describe('Sign Typed Data Signature Request', function () {
```

Comment:
I'm not sure I understand the purpose of these duplicated tests?

My assumption was we'd update the default fixtures to disable petnames by default meaning all the existing E2E tests run as they did previously.

And then our explicit petnames tests such as `petnames-signatures.spec.js` are used to verify the petnames logic specifically.

Then in future when it becomes enabled by default, we'd change the fixture default and update any tests as needed.

Reply:
Matt, the tests for the 'petnames' seem to focus mainly on the saving functionality. While this is important, I felt we might be missing some other crucial aspects, such as the ability to send eth and sign requests when the petnames flag is enabled.

I'm not entirely sure if these scenarios are covered in our current testing suite. If they aren't, I believe it would be beneficial to include them to ensure our feature is robust and reliable.

Reply:
As we're not enabling by default, we can default it on in the fixtures to test everything with the name component in place.

For the sake of time, if we hit any errors that rely on the old component, we can change those specific tests to set the toggle off and address them in a separate PR?

This way we shouldn't need any duplicate tests since petnames will be implicitly tested in all the other tests?

Reply:
Done.